### ISSUE TITLE
CA1822 with api_surface=private triggers on public field of private class

### ISSUE THREAD

**Issue by shuebner** [dotnet/roslyn-analyzers#6665]
### Analyzer

**Diagnostic ID**: [CA1822](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 7.0.203

_OR_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.1 (Latest)

### Describe the bug

CA1822 with `api_surface` set to `private` is triggered on public members of private classes.

### Steps To Reproduce

code:
```csharp
public class Class1
{
    private class PrivateClass
    {
        public string Value => "value";
    }
}
```

`.globalconfig`:
```editorconfig
dotnet_code_quality.CA1822.api_surface = private
```

### Expected behavior

CA1822 does not trigger because the member in question (`Value` in the above example) is `public`, regardless of the class visibility.

### Actual behavior

CA1822 triggers.

## Additional context

I could not find any rules for `api_surface` in the documentation for cases like this.

**Comment by mavasani** [comment]
`api_surface` option operates based on the effective visibility of the symbol. However, it seems reasonable to expect that a value `private` does not trigger for any non-private symbols. This may be a breaking change though, so I am wondering if a better approach here is to add another parallel option, say `declared_accessibility` which only matches the accessibility modifier applied directly to the symbol. Or maybe the breaking change to `api_surface` is acceptable. This would definitely need some design discussion.

**Comment by m-gallesio** [comment]
Found this today for `public` members on `internal` classes. Having two parallel options seems the most flexible solution.
